ksqldb-common/src/main/java/io/confluent/ksql/properties/DenyListPropertyValidator.java

```diff
@@ -32,8 +32,8 @@
   private final Set<String> immutableProps;
 
   public DenyListPropertyValidator(final Collection<String> immutableProps) {
-    this.immutableProps = ImmutableSet.copyOf(
-        Objects.requireNonNull(immutableProps, "immutableProps"));
+    this.immutableProps = ImmutableSet.<String>builder().addAll(
+        Objects.requireNonNull(immutableProps, "immutableProps")).add("ksql.service.id").build();
```

Comment:
I think you should be able to specify `KsqlConfig.KSQL_SERVICE_ID_CONFIG` here instead of hardcoding the property